visualizer/actionButton.js

```diff
@@ -223,44 +332,22 @@
         if (!argNames || argNames.length === 0) {
           wrapper.querySelector('.buttonWrapper').hidden = true;
         } else {
-          wrapper.querySelector('.buttonWrapper').querySelector('.add').hidden = true;
-          wrapper.querySelector('.buttonWrapper').querySelector('.arrow').hidden = false;
+          wrapper.querySelector('.buttonWrapper .add').hidden = true;
+          wrapper.querySelector('.buttonWrapper .arrow').hidden = false;
         }
       }
       // TODO: add operation/attribute to force evaluation menu
       try {
-        ohmEditor.semantics.emit('add:semanticOperation', type, name, args);
+        var origActionDict = wrapper._origActionDict;
```

Comment:
It would be good to move everything that can't throw an error outside of the `try` block. That way it's very clear where the error can come from (and gives a better idea of what kind of errors are possible).